File: codewars/7kyu/substring-fun/dduck-substring-fun.py
# https://www.codewars.com/kata/substring-fun/train/javascript

nth_char = lambda words: "".join([words[i][i] for i in range(len(words))])

# const nth_char = (strArr, str = '', count = 0) => {
#   return strArr.length ? nthChar(strArr, str += strArr.shift()[count] || '', count + 1) : str;
# };

print(nth_char(['yoda', 'best', 'has']))
# 'yes');
# print(nth_char(['']))

File: codewars/7kyu/substring-fun/dduck-substring-fun.js
// https://www.codewars.com/kata/substring-fun/train/javascript

const nthChar = (strArr, str = '', count = 0) => {
  return strArr.length ? nthChar(strArr, str += strArr.shift()[count] || '', count + 1) : str;
};

// nthChar(['yoda', 'best', 'has']);
//?
nthChar(['']);
//?
//, 'yes');
